### PR TITLE
WIP: customizer all the things

### PR DESCRIPTION
**Pull request by adriancole**
This adds hooks for new customizers that will be in Brave 5.7. As I'll
be offline for a day or two, if anyone wants to play with this they can.

We could just weave-in and document this, or we could retrofit some of
our existing stuff to configure itself via customizers instead (ex MDC
integration).

cc @devinsba

### CODE REVIEW

**Comment by marcingrzejszczak** [comment]
Lgtm

I'd refactor what we have to use the customizers I guess

**Comment by narayaruna** [comment]
@marcingrzejszczak Netflix needs this. Could you include it in 2.1.3. Thanks.

@adriancole fyi

**Comment by shakuzen** [comment]
Brave 5.7 is being released now.

**Comment by adriancole** [comment]
this is now on a stable release. needs tests and docs.. not sure if you have time for it @marcingrzejszczak but I'm offline until monday. hoping to get this in (also 2.1) to unlock the secondary sampling thing nara's hoping to roll out.

File: spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/autoconfig/TraceAutoConfiguration.java [hunk 1]
```diff
@@ -107,6 +119,9 @@ Tracing tracing(
 		for (FinishedSpanHandler finishedSpanHandlerFactory : this.finishedSpanHandlers) {
 			builder.addFinishedSpanHandler(finishedSpanHandlerFactory);
 		}
+		for (TracingCustomizer customizer : this.tracingCustomizers) {
```
Comment (adriancole) [inline spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/autoconfig/TraceAutoConfiguration.java]: ps without tests this is subtle. it is important these apply last as they can rejig the reporter config, something needed for secondary sampling to work. ex we want to flip default reporter to noop but reuse the normal reporter as a finished span handler. this stuff works when customizes hit last before build

**Comment by adriancole** [comment]
bravo!

**Pull request closed without merging**